compiler/rustc_session/src/parse.rs

```diff
@@ -217,6 +217,11 @@ pub struct ParseSess {
     pub proc_macro_quoted_spans: AppendOnlyVec<Span>,
     /// Used to generate new `AttrId`s. Every `AttrId` is unique.
     pub attr_id_generator: AttrIdGenerator,
+    /// Track parse errors that will cause unnecessary resolve errors.  This is a crude mechanism,
+    /// we should do something that is scope-based, instead of crate-global.
+    pub silence_resolve_errors: Lock<bool>,
```

Comment:
This needs to take an `ErrorGuaranteed`. Currently it seems way too easy to silence resolver errors without making sure at least one error has been emitted.

Reply:
Actually, since it's being delayed later on, whatever. But I do think it should still take an `ErrorGuaranteed`, if only to ensure that we emit a parser error in the right place.